pkg/internal/controllers/certificaterequestpolicies.go

```diff
@@ -118,6 +119,9 @@ func addCertificateRequestPolicyController(_ context.Context, opts Options) erro
 
 	return ctrl.NewControllerManagedBy(opts.Manager).
 		For(new(policyapi.CertificateRequestPolicy)).
+		// Reconcile the object only if there is a change in the spec or label.
+		// This prevents an infinite reconcile loop when the status is updated.
+		WithEventFilter(predicate.Or(predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{})).
```

Comment:
Why do we need events when policy labels are changed?

Reply:
I wanted to cover possible future cases, but you are right, it's not need, will remove it.

Reply:
As agreed with Tim, I've removed predicate since problem was related with updating `LastTransitionTime` and setting ns for cluster wide object: ` b.WithNamespace(namespace)`.